### PR TITLE
[CLI][e2e] Tests create-resource-account and derive-resource-account-address

### PR DESCRIPTION
**Pull request by 0xjinn**
### Description

Add e2e for testing

1. create-resource-account with --seed
2. derive-resource-account-address with the same --seed value

### Test Plan

```
➜  e2e git:(main) ✗ DOCKER_DEFAULT_PLATFORM=linux/amd64 poetry run python main.py --base-network testnet --test-cli-path ~/Projects/aptos-core/target/debug/aptos
Configuration file exists at /Users/jinhou/Library/Preferences/pypoetry, reusing this directory.

Consider moving TOML configuration files to /Users/jinhou/Library/Application Support/pypoetry, as support for the legacy directory will be removed in an upcoming release.
2023-06-20 14:35:02,651 - INFO - Trying to run aptos CLI local testnet from image: aptoslabs/tools:testnet
testnet: Pulling from aptoslabs/tools
Digest: sha256:e2cd216dcd7225546ba47caadd8d6cbe34508028988eba474145bd3032320c15
Status: Image is up to date for aptoslabs/tools:testnet
2023-06-20 14:35:04,412 - INFO - Running aptos CLI local testnet from image: aptoslabs/tools:testnet
2023-06-20 14:35:04,412 - INFO - Waiting for node and faucet APIs for aptos-tools-testnet to come up
2023-06-20 14:35:20,716 - INFO - Node and faucet APIs for aptos-tools-testnet came up
2023-06-20 14:35:20,732 - INFO - Running test: test_metrics_accessible
2023-06-20 14:35:20,811 - INFO - Running test: test_init
2023-06-20 14:35:23,160 - INFO - Running test: test_account_fund_with_faucet
2023-06-20 14:35:24,245 - INFO - Running test: test_account_create
2023-06-20 14:35:27,082 - INFO - Running test: test_account_lookup_address
2023-06-20 14:35:28,002 - INFO - Running test: test_account_resource_account
2023-06-20 14:35:31,778 - INFO - Running test: test_aptos_header_included
2023-06-20 14:35:31,820 - INFO - Running test: test_move_publish
2023-06-20 14:35:41,621 - INFO - Stopping container: aptos-tools-testnet
2023-06-20 14:35:41,883 - INFO - Stopped container: aptos-tools-testnet
2023-06-20 14:35:41,883 - INFO - These tests passed:
2023-06-20 14:35:41,883 - INFO - test_metrics_accessible
2023-06-20 14:35:41,883 - INFO - test_init
2023-06-20 14:35:41,883 - INFO - test_account_fund_with_faucet
2023-06-20 14:35:41,883 - INFO - test_account_create
2023-06-20 14:35:41,883 - INFO - test_account_lookup_address
2023-06-20 14:35:41,883 - INFO - test_account_resource_account
2023-06-20 14:35:41,883 - INFO - test_aptos_header_included
2023-06-20 14:35:41,883 - INFO - test_move_publish
2023-06-20 14:35:41,883 - INFO - All tests passed!
```

### CODE REVIEW

File: crates/aptos/e2e/cases/account.py [hunk 1]
```diff
@@ -80,3 +82,48 @@ def test_account_lookup_address(run_helper: RunHelper, test_name=None):
...
+            seed,
+            "--assume-yes",  # assume yes to gas prompt
+        ],
+    )
+
+    result = json.loads(result.stdout)
+    sender = result["Result"]["sender"]
+    resource_account_address = result["Result"]["resource_account"]
+
+    if resource_account_address == None or sender == None:
```
Comment (banool) [inline crates/aptos/e2e/cases/account.py:109]: Are we sure this actually works as intended when resource account is none? I figure the `resource_account` key won't be present. If so, we'll get a KeyError. Do we need this instead?

```suggestion
    resource_account_address = result["Result"].get("resource_account")

    if resource_account_address == None or sender == None:
```

Same thing with sender.

File: crates/aptos/e2e/cases/account.py [hunk 2]
```diff
@@ -80,3 +82,48 @@ def test_account_lookup_address(run_helper: RunHelper, test_name=None):
...
+        raise TestError("resource account creation failed")
```
Comment (banool) [inline crates/aptos/e2e/cases/account.py:110]:
```suggestion
        raise TestError("Resource account creation failed")
```

File: crates/aptos/e2e/cases/account.py [hunk 3]
```diff
@@ -80,3 +82,48 @@ def test_account_lookup_address(run_helper: RunHelper, test_name=None):
         raise TestError(
             f"lookup-address result does not match {run_helper.get_account_info().account_address}"
         )
+
+
+@test_case
+def test_account_resource_account(run_helper: RunHelper, test_name=None):
```
Comment (banool) [inline crates/aptos/e2e/cases/account.py:88]: Could you make a call to the local testnet to confirm that the resource account was actually created on chain?

**Review by banool — CHANGES_REQUESTED** [review]

File: crates/aptos/e2e/cases/account.py (hunk 1 above)
Comment (0xjinn) [reply crates/aptos/e2e/cases/account.py:109]: Hmm I thought both of them means the same thing. TIL!

**Comment by 0xjinn** [comment]
Updated to use `aptos account list` to make sure resource been created

**Pull request merged**